src/HttpAsyncClientDiscovery.php

```diff
@@ -24,7 +24,7 @@ public static function find()
         try {
             $asyncClient = static::findOneByType(HttpAsyncClient::class);
 
-            return new $asyncClient();
+            return static::instantiateClass($asyncClient);
```

Comment:
I don't understand how the linked method is relevant: the exceptions there would still not be caught by this catch block.

Yes, I propose exactly that.